KarhooUISDK/Screens/CheckoutScreen/Checkout/KarhooCheckoutViewController.swift

```diff
@@ -36,10 +36,11 @@ final class KarhooCheckoutViewController: UIViewController, CheckoutView {
     private var termsConditionsView: TermsConditionsView!
     private var containerBottomConstraint: NSLayoutConstraint!
     private let drawAnimationTime: Double = 0.45
+    private let shouldDisplayLegalNotice: Bool = UITexts.Booking.legalNoticeText.isNotEmpty
     var presenter: CheckoutPresenter
     var passengerDetailsValid: Bool?
     var paymentNonce: String?
-
+    private var legalNoticeLinkOperer: LegalNoticeLinkOpener!
```

Comment:
Since instance is assigned to this variable in the initialization step, there is no need to use optional type.

Reply:
it's still required because I need to call super.init first, before I initialize legalNoticeLinkOpener